pkg/fanal/analyzer/language/c/conan/conan.go

```diff
@@ -1,42 +1,148 @@
-	p := conan.NewParser()
-	res, err := language.Analyze(types.Conan, input.FilePath, input.Content, p)
+func (a conanLockAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysisInput) (*analyzer.AnalysisResult, error) {
+	required := func(filePath string, d fs.DirEntry) bool {
+		return a.Required(filePath, nil)
+	}
+
+	licenses, err := licensesFromCache()
 	if err != nil {
-		return nil, xerrors.Errorf("%s parse error: %w", input.FilePath, err)
+		a.logger.Debug("Unable to parse cache directory to obtain licenses", log.Err(err))
+	}
+
+	var apps []types.Application
+	if err = fsutils.WalkDir(input.FS, ".", required, func(filePath string, _ fs.DirEntry, r io.Reader) error {
+		app, err := language.Parse(types.Conan, filePath, r, a.parser)
+		if err != nil {
+			return xerrors.Errorf("%s parse error: %w", filePath, err)
+		}
+
+		if app == nil {
+			return nil
+		}
+
+		// Fill licenses
+		for i, lib := range app.Libraries {
+			if license, ok := licenses[lib.Name]; ok {
+				app.Libraries[i].Licenses = []string{
+					license,
+				}
+			}
+		}
+
+		sort.Sort(app.Libraries)
+		apps = append(apps, *app)
+		return nil
+	}); err != nil {
+		return nil, xerrors.Errorf("unable to parse conan lock file: %w", err)
+	}
+
+	return &analyzer.AnalysisResult{
+		Applications: apps,
+	}, nil
+}
+
+func licensesFromCache() (map[string]string, error) {
+	required := func(filePath string, d fs.DirEntry) bool {
+		return filepath.Base(filePath) == "conanfile.py"
+	}
+
+	// cf. https://docs.conan.io/1/mastering/custom_cache.html
+	cacheDir := os.Getenv("CONAN_USER_HOME")
+	if cacheDir == "" {
+		cacheDir, _ = os.UserHomeDir()
+	}
+	cacheDir = path.Join(cacheDir, ".conan", "data")
+
+	if !fsutils.DirExists(cacheDir) {
+		return nil, xerrors.Errorf("the Conan cache directory (%s) was not found.", cacheDir)
+	}
+
+	licenses := make(map[string]string)
+	if err := fsutils.WalkDir(os.DirFS(cacheDir), ".", required, func(filePath string, _ fs.DirEntry, r io.Reader) error {
+		scanner := bufio.NewScanner(r)
+		var name, license string
+		for scanner.Scan() {
+			line := strings.TrimSpace(scanner.Text())
+
+			if strings.HasPrefix(line, "name") { // cf. https://docs.conan.io/1/reference/conanfile/attributes.html#name
+				// trim extra characters - e.g. `name = "openssl"` -> `openssl`
+				name = strings.TrimSuffix(strings.TrimPrefix(line, `name = "`), `"`)
+				// Check that the license is already found
+				if license != "" {
+					break
+				}
+			} else if strings.HasPrefix(line, "license") { // cf. https://docs.conan.io/1/reference/conanfile/attributes.html#license
+				// trim extra characters - e.g. `license = "Apache-2.0"` -> `Apache-2.0`
+				license = strings.TrimSuffix(strings.TrimPrefix(line, `license = "`), `"`)
+				// Check that the name is already found
+				if name != "" {
+					break
+				}
+			}
+		}
+
+		// Skip files without name/license
+		if name == "" || license == "" {
+			return nil
+		}
```

Comment:
We also need to parse the version as the license can be different depending on the version.

Reply:
if i understand correctly - `conan` uses same file for each version:
repository:
```bash
➜  conan wget -q https://github.com/conan-io/conan-center-index/raw/master/recipes/zlib/all/conanfile.py
➜  conan shasum conanfile.py
2a864cb78aacb2d27818be9c109e4f3fc1293b3a  conanfile.py
```
cache dir:
```bassh
root@3bc66c3cd5d8:/examples2/tutorial/consuming_packages/simple_cmake_project# sha1sum /root/.conan2/p/zlib2cad50ee1214e/e/conanfile.py
2a864cb78aacb2d27818be9c109e4f3fc1293b3a  /root/.conan2/p/zlib2cad50ee1214e/e/conanfile.py
root@3bc66c3cd5d8:/examples2/tutorial/consuming_packages/simple_cmake_project# sha1sum /root/.conan2/p/zlib345774da50bf7/e/conanfile.py
2a864cb78aacb2d27818be9c109e4f3fc1293b3a  /root/.conan2/p/zlib345774da50bf7/e/conanfile.py
```

Repository also contains 1 conanfile.py file for all versions. Look at `conandata.yaml` file - https://github.com/conan-io/conan-center-index/blob/master/recipes/zlib/all/conandata.yml.
 
I was thinking about the case where different versions use different licenses. But I didn't find an example for this.

So I think we can define licenses by package name first.
If we have any problems, we will update this logic.

Reply:
>Repository also contains 1 conanfile.py file for all versions. Look at conandata.yaml file - https://github.com/conan-io/conan-center-index/blob/master/recipes/zlib/all/conandata.yml.

Where is the `conanfile.py` file defined in `conandata.yaml`? I found URLs, but didn't find `conanfile.py`.

Reply:
if i understand correctly here it is  -https://github.com/conan-io/conan-center-index/blob/master/recipes/zlib/all/conanfile.py

Reply:
I probably got you. There is only one file stored in the repository.
https://github.com/conan-io/conan-center-index/blob/258419698ea54b9bfd86a10f786e72759af647c2/recipes/zlib/all/conanfile.py

It seems like conan is unable to define different licenses. We can go with you implementation.